go.mod

```diff
@@ -1,82 +1,59 @@
+	github.com/golang/protobuf v1.3.2
+	github.com/google/btree v1.0.0 // indirect
+	github.com/imdario/mergo v0.3.7 // indirect
 	github.com/opencontainers/go-digest v1.0.0-rc1 // indirect
-	github.com/pborman/uuid v0.0.0-20170612153648-e790cca94e6c // indirect
-	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
-	github.com/prometheus/client_golang v0.8.0
-	github.com/prometheus/client_model v0.0.0-20180712105110-5c3871d89910 // indirect
-	github.com/prometheus/common v0.0.0-20180801064454-c7de2306084e // indirect
-	github.com/prometheus/procfs v0.0.0-20180725123919-05ee40e3a273 // indirect
-	github.com/sirupsen/logrus v1.3.0 // indirect
-	github.com/soheilhy/cmux v0.1.4 // indirect
-	github.com/spf13/afero v1.2.1 // indirect
+	github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
 	github.com/spf13/cobra v0.0.3
-	github.com/spf13/pflag v1.0.2
-	github.com/stretchr/testify v1.3.0 // indirect
-	github.com/tmc/grpc-websocket-proxy v0.0.0-20190109142713-0ad062ec5ee5 // indirect
-	github.com/ugorji/go v1.1.1 // indirect
-	github.com/vmware/govmomi v0.20.0
-	github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2 // indirect
-	go.etcd.io/bbolt v1.3.2 // indirect
-	golang.org/x/net v0.0.0-20181220203305-927f97764cc3
-	golang.org/x/sync v0.0.0-20190227155943-e225da77a7e6 // indirect
-	golang.org/x/time v0.0.0-20180412165947-fbb02b2291d2 // indirect
-	google.golang.org/grpc v1.19.0
+	github.com/spf13/pflag v1.0.3
+	github.com/vmware/govmomi v0.21.0
+	golang.org/x/net v0.0.0-20190724013045-ca1201d0de80
+	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4 // indirect
+	google.golang.org/grpc v1.22.1
 	gopkg.in/gcfg.v1 v1.2.3
-	gopkg.in/inf.v0 v0.9.1 // indirect
-	gopkg.in/natefinch/lumberjack.v2 v2.0.0-20170531160350-a96e63847dc3 // indirect
-	gopkg.in/square/go-jose.v2 v2.1.8 // indirect
+	gopkg.in/square/go-jose.v2 v2.3.1 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
-	k8s.io/api v0.0.0-20180628040859-072894a440bd
-	k8s.io/apiextensions-apiserver v0.0.0-20180822171001-b12c11a9bd71 // indirect
-	k8s.io/apimachinery v0.0.0-20180621070125-103fd098999d
-	k8s.io/apiserver v0.0.0-20180628044425-01459b68eb5f
-	k8s.io/client-go v8.0.0+incompatible
-	k8s.io/klog v0.2.0
-	k8s.io/kube-openapi v0.0.0-20180731170545-e3762e86a74c // indirect
-	k8s.io/kubernetes v1.11.2
-	k8s.io/sample-controller v0.0.0-20180822125000-be98dc6210ab
+	k8s.io/api v0.0.0
+	k8s.io/apimachinery v0.0.0
+	k8s.io/client-go v0.0.0
+	k8s.io/cloud-provider v0.0.0
+	k8s.io/component-base v0.0.0
+	k8s.io/klog v0.3.2
+	k8s.io/kube-openapi v0.0.0-20190401085232-94e1e7b7574c // indirect
+	k8s.io/kubernetes v1.15.0
+	k8s.io/sample-controller v0.0.0-20190731144349-6f8905ae4ee5
+)
+
+replace (
+	// these replacements are pinned to e8462b5b5dc2 which is the sha associated with the 1.15.0 tag on k/k
+	// as you cannot pin them to v1.15.0 directly
+	k8s.io/api => k8s.io/kubernetes/staging/src/k8s.io/api v0.0.0-20190806130100-e8462b5b5dc2
```

Comment:
I think the pseudoversion here needs to be `v0.0.0-20190615005809-e8462b5b5dc2`.

The error from https://proxy.golang.org/k8s.io/kubernetes/staging/src/k8s.io/cli-runtime/@v/v0.0.0-20190806130100-e8462b5b5dc2.info:

```
not found: k8s.io/kubernetes/staging/src/k8s.io/cli-runtime@v0.0.0-20190806130100-e8462b5b5dc2: invalid pseudo-version: does not match version-control timestamp (2019-06-15T00:58:09Z)
```

The pseudoversion can be found by: `TZ=UTC git show -q --pretty='format:v0.0.0-%cd-%h' --date='format-local:%Y%m%d%H%M%S' --abbrev=12`

Reply:
thanks @nikhita 🙏